### PR TITLE
Renaming launcher executable to `ensoup`

### LINKED ISSUE #10476: Avoid name duplication of the `enso` universal launcher
There are two executables named `enso` among the daily bits we produce. That's **confusing** and _we shall fix it_.

One of them is called **Enso launcher**. To learn about it, please visit our download page: https://github.com/enso-org/enso/releases/tag/2024.1.1 - the download is in section called **Enso Engine** on that page, but in fact, it is **not the engine**, it is **the launcher**!

Enso Engine

To learn more abou tthe **enso launcher** visit this [documentation page](https://github.com/enso-org/enso/blob/d95ebd483684f270cb1ee04ae126d3be3560b086/docs/distribution/distribution.md) - it contains nice description of the whole **universtal launcher/updater** picture...

However to benefit from the work done by 
- #10374 
- and to fulfill the vision as discussed in #10121

we need to find a _better location_ for exposing _native image_ built launcher. The idea is to **rename the launcher** to `ensoup`:

ensoup illustration

It is a nice, _illustrative_ name derived from the `rustup` example that nicely describes one of the purposes of `ensoup` - download the distributions of Enso automatically. Once the `enso` in the bundle is renamed to `ensoup`, we can proceed with placing a _native image built_ binary next to it under the name `enso` - but that's for a different PR.

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Closes #10476.

### Important Notes

Let's see what exactly fails on the CI and fix that then...

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been updated where possible.
- [x] Rename `OS.executableName("enso")` calls to `ensoup`

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
CI is green. However, how do I verify the release is OK too? Can I somehow download `enso-bundle` and verify `ensoup` is there? CCing @radeusgd

**Comment by radeusgd** [comment]
> CI is green. However, how do I verify the release is OK too? Can I somehow download `enso-bundle` and verify `ensoup` is there? CCing @radeusgd

I'm not sure if I know any better, the release workflows have changed heavily since I was touching them ages ago. I imagine the only way to find out is to run the release workflow and check the results. I think just running will not yet publish the release but it should only prepare a draft (but probably @jdunkerley or @AdRiley will have better knowledge of that). We could test such a draft and then remove it.

**Comment by radeusgd** [comment]
There is suprisingly little changes.

It may be a minimal set to keep things working, but I think we should change more.

For example, there's many `OS.executableName("enso")` calls in the `ensoup` launcher self-update logic. This is just temp files so their names are not really too important, but I think renaming them to `ensoup` will allow us to avoid confusion when debugging stuff (if something breaks and we will be analyzing, we will get confused why a file called `enso` is actually `ensoup`). So IMO would be good to align those.

**Comment by radeusgd** [comment]
Also, please take a look at `InstallerSpec.scala` - test `install itself`. If it is passing with current state, that means that after I call `./path/to/downloaded/ensoup install distribution`, the `ensoup` updater gets renamed back to `enso` (and worse, it suggests to put it on user's PATH so it will clash with any engine executables called `enso` that could be on the path). That's bad. If we want to use the `ensoup` binary name, let's stick to it consistently.

Look into `DistributionManager.executableName` - it's the launcher/updater executable name. If we apply the above change (renaming calls to `OS.executableName` that should get fixed by the way).

**Comment by JaroslavTulach** [comment]
> > However, how do I verify the release is OK too? 
> 
> I'the only way to find out is to run the release workflow and check the results. 

Running the [nightly release](https://github.com/enso-org/enso/actions/runs/9907875988) for this branch now.

**Comment by hubertp** [comment]
Have you tried `./run ide2 build --skip-version-check` to build/run resulting AppImage?

**Comment by JaroslavTulach** [comment]
> Running the [nightly release](https://github.com/enso-org/enso/actions/runs/9907875988) for this branch now.

Fails with [following message](https://github.com/enso-org/enso/actions/runs/9907875988/job/27373183811#step:7:1570):
```
java.lang.IllegalArgumentException: requirement failed: Source file '/runner/_work/enso/enso/enso' does not exist.
```
The failing command is [sbt "all engine-runner/assembly buildEngineDistribution buildProjectManagerDistribution buildLauncherDistribution](https://github.com/enso-org/enso/actions/runs/9907875988/job/27373183811#step:7:1570).

Fixed in 115fa77 and a [night release run](https://github.com/enso-org/enso/actions/runs/9911231885) scheduled again.

None of the artifacts seems to contain `ensoup`! I see only `project-manager`, engine and GraalVM:
```
$ tar ftvz ide-unpacked-linux.tar.gz ./ | grep /bin/ | grep -v graalvm
drwxr-xr-x ci/ci             0 2024-07-12 18:59 ./resources/enso/bin/
-rwxr-xr-x ci/ci     215146400 2024-07-12 18:59 ./resources/enso/bin/project-manager
drwxr-xr-x ci/ci             0 2024-07-12 18:59 ./resources/enso/dist/2024.3.1-nightly.2024.7.12.1/bin/
-rwxr-xr-x ci/ci           389 2024-07-12 18:59 ./resources/enso/dist/2024.3.1-nightly.2024.7.12.1/bin/enso
-rw-r--r-- ci/ci           435 2024-07-12 18:51 ./resources/enso/dist/2024.3.1-nightly.2024.7.12.1/bin/enso.bat
```
Looks like `AppImage` isn't using _Enso Universal Launcher_ at all!

**Comment by JaroslavTulach** [comment]
> or example, there's many `OS.executableName("enso")` calls in the `ensoup` launcher self-update logic. This is just temp files so their names are not really too important, but I think renaming them to `ensoup` will allow us to avoid confusion

Attempted to address this comment in e2bae49ccd1acaaeb083835bc6b8ea62411901e7

**Comment by JaroslavTulach** [comment]
> Have you tried `./run ide2 build --skip-version-check` to build/run resulting AppImage?

AppImage doesn't contain `ensoup` at all.

**Comment by JaroslavTulach** [comment]
> Also, please take a look at `InstallerSpec.scala` - test `install itself`. 
> Look into `DistributionManager.executableName` - it's the launcher/updater executable name. 

Fixed in 889fae8

**Comment by JaroslavTulach** [comment]
I've just [updated the documentation](https://github.com/enso-org/enso/pull/10535/commits/b1336536a182ff0ac194fd1e154e43d574312200) a bit. The documentation still refers to _launcher_ at various places - let it be. At least the layout describes there is `bin/ensoup`.

**Review by radeusgd — APPROVED** [review]

File: engine/launcher/src/test/scala/org/enso/launcher/installation/InstallerSpec.scala [hunk 1]
```diff
@@ -66,9 +66,9 @@ class InstallerSpec extends NativeTest with WithTemporaryDirectory {
         env
       )
 
-      (installedRoot / "bin" / OS.executableName("enso")).toFile should exist
+      (installedRoot / "bin" / OS.executableName("ensoup")).toFile should exist
```
Comment (hubertp) [inline engine/launcher/src/test/scala/org/enso/launcher/installation/InstallerSpec.scala]:
```suggestion
      (installedRoot / "bin" / OS.executableName(Constants.name)).toFile should exist
```

File: engine/launcher/src/test/scala/org/enso/launcher/installation/InstallerSpec.scala [hunk 2]
```diff
@@ -66,9 +66,9 @@ class InstallerSpec extends NativeTest with WithTemporaryDirectory {
...
       assert(
-        Files.isExecutable(installedRoot / "bin" / OS.executableName("enso")),
+        Files.isExecutable(installedRoot / "bin" / OS.executableName("ensoup")),
```
Comment (hubertp) [inline engine/launcher/src/test/scala/org/enso/launcher/installation/InstallerSpec.scala]:
```suggestion
        Files.isExecutable(installedRoot / "bin" / OS.executableName(Constants.name)),
```

File: engine/launcher/src/test/scala/org/enso/launcher/installation/InstallerSpec.scala [hunk 3]
```diff
@@ -97,7 +97,7 @@ class InstallerSpec extends NativeTest with WithTemporaryDirectory {
         env
       )
 
-      (installedRoot / "bin" / OS.executableName("enso")).toFile should exist
+      (installedRoot / "bin" / OS.executableName("ensoup")).toFile should exist
```
Comment (hubertp) [inline engine/launcher/src/test/scala/org/enso/launcher/installation/InstallerSpec.scala]:
```suggestion
      (installedRoot / "bin" / OS.executableName(Constants.name)).toFile should exist
```

**Review by hubertp — APPROVED** [review]
It would probably make sense to have some plain subproject with constants only. That way we wouldn't repeat that all over the place.

**Comment by JaroslavTulach** [comment]
Let's get this in to see the overall consequences (I still don't know who builds the `engine-bundle` artifacts).

**Review by Akirathan — APPROVED** [review]

**Review by kazcw — APPROVED** [review]

**Pull request merged**